dissect/target/plugins/general/users.py

```diff
@@ -82,8 +82,49 @@ def all(self) -> Iterator[UserDetails]:
 
     def all_with_home(self) -> Iterator[UserDetails]:
         """Return :class:`UserDetails` objects for users that have existing directory set as home directory."""
-        for user in self.target.users():
-            if user.home:
-                user_details = self.get(user)
-                if user_details.home_path.exists():
-                    yield user_details
+
+        seen: set[TargetPath] = set()
+        try:
+            for user in self.target.users():
+                if user.home:
+                    user_details = self.get(user)
+                    if user_details.home_path.exists():
+                        seen.add(user_details.home_path)
+                        yield user_details
+        except Exception as e:
+            self.target.log.warning("Failed to retrieve user details, falling back to hardcoded locations.")
+            self.target.log.debug("", exc_info=e)
+
+        # Iterate over misc home directories
+        for misc_home_dir, user_criterion in self.target.misc_home_dirs():
+            # We skip the entry if no user can be found.
+            # We cannot use set membership check here because of https://github.com/fox-it/dissect.target/issues/1203
+            if any(seen_path.samefile(misc_home_dir) for seen_path in seen) or not user_criterion:
+                continue
+
+            if (user := self.find(**{user_criterion[0]: user_criterion[1]})) is None:
+                continue
+
+            yield UserDetails(user=UnixUserRecord(name=misc_home_dir), home_path=misc_home_dir)
```

Comment:
Maybe introduce a barebones `UserRecord`?

Reply:
Not sure what you mean, but setting user this way was a bug, we have a user record from `self.find`.

Amended unit test

Reply:
I meant to not abuse `UnixUserRecord` even though we may not be running on a Unix system, but it's irrelevant now.